src/main/java/hillel/doctorRest/clinic/schedule/ScheduleController.java

```diff
@@ -74,12 +72,45 @@ public Schedule createSchedule(@PathVariable Integer doctorId,
             throw new PetNotFoundException();
         } else if (!visitHoursConfig.getHourName().contains(hour)) {
             throw new HourNotFoundException();
-        } else if (scheduleService.findByHour(hour).isPresent()) {
+        } else if (!scheduleService.findByDoctorIdAndVisitDateAndHour(doctorId, visitDate, hour).isEmpty()) {
             throw new HourNotFoundAvailableException();
         } else {
             return scheduleService
                     .createSchedule(doctorId, visitDate, hour, scheduleDtoConverter
                             .toModel(scheduleInputDto));
         }
     }
+
+    public String idDoc(Integer idDoc) {
+        return "Schedule docId =" + idDoc.toString();
+
+    }
+
+    public Map<String, String> sortedMapToPetId(Map<String, String> map) {
+        return map.entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue,
+                        (oldValue, newValue) -> oldValue, LinkedHashMap::new));
```

Comment:
It is unclear what this method is expected to do. What are the keys in input 'map'?  What are the values there?
Looks like the merge function is incorrect (unless it is intended to drop values for repeated keys).

As I see from the method name, you just wanted to sort the map by key. If it is the only intention, so more straight forward way is to use `TreeMap` for that